### PR TITLE
[DNM][stdlib] Add exponentiation precedence group and operator for diagnostic purposes

### PR DESCRIPTION
**Pull request by xwu**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by xwu** [comment]
@swift-ci smoke test

**Comment by xwu** [comment]
@swift-ci smoke test

**Comment by xwu** [comment]
cc @varungandhi-apple

File: stdlib/public/core/Policy.swift [hunk 1]
```diff
@@ -486,3 +491,48 @@ infix operator   |=: AssignmentPrecedence, BinaryInteger
...
+extension Float80 {
+  @available(*, unavailable, message: "use 'Foundation.pow' instead")
+  @_alwaysEmitIntoClient
+  public static func ** (lhs: Self, rhs: Self) -> Self {
+    fatalError("Not implemented")
+  }
+  @available(*, unavailable, message: "use 'Foundation.pow' instead")
+  @_alwaysEmitIntoClient
+  public static func **= (lhs: inout Self, rhs: Self) {
+    fatalError("Not implemented")
```
Comment (benrimmington) [inline stdlib/public/core/Policy.swift:535]: I'm not sure if you can use `fatalError` within the standard library.
Can you use `_preconditionFailure()` or `Builtin.unreachable()` instead?
e.g. [FloatingPointTypes.swift.gyb](https://github.com/apple/swift/blob/main/stdlib/public/core/FloatingPointTypes.swift.gyb) — `isSignMinus` and `quietNaN`.
e.g. [UnavailableStringAPIs.swift](https://github.com/apple/swift/blob/main/stdlib/public/core/UnavailableStringAPIs.swift) — `String` subscripting with integers.

There are also `pow` functions for `CGFloat` and `Decimal`.
Can the unavailable APIs be added to a `Numeric` protocol extension?
(The error message would need to be more general.)

Most `pow` functions use a single type, i.e. `(Self, Self) -> Self`.
But the [`Decimal`](https://developer.apple.com/documentation/foundation/1779833-pow) API uses an `Int` power.
Should the unavailable APIs have a generic other type? i.e. `(Self, Other) -> Self`.

Comment (xwu) [reply stdlib/public/core/Policy.swift:535]: I've deliberately scoped the placeholders to those `pow` functions that exist in the C standard library; I'm not satisfied with the wording of the message and will likely revise to call out the C standard library specifically ("import Foundation or the C standard library to use 'pow' instead").

If/when Swift Numerics gets merged into the standard library, then `**` implementations can be made generic over `Real` to call `Real.pow`. But `CGFloat` belongs to CoreGraphics, and `Decimal` to Foundation, and those framework owners get to decide how their APIs are exposed and whether or not they want to participate in conformance to `Real` and/or use the `**` operator.

Comment (xwu) [reply stdlib/public/core/Policy.swift:535]: > I'm not sure if you can use `fatalError` within the standard library.

I've both used and removed (when implementing previously undone features) plenty of `fatalError`s within the standard library. It's the standard idiom for these things (just scroll to the bottom of the `FloatingPointTypes.swift.gyb` file that you link to).

Comment (benrimmington) [reply stdlib/public/core/Policy.swift:535]: I was thinking of `precondition` versus `_precondition` usage.

**Comment by xwu** [comment]
@swift-ci smoke test macOS

**Comment by xwu** [comment]
Closing in favor of #36074.

**Pull request closed without merging**